CONTRIBUTING.md

```diff
@@ -0,0 +1,89 @@
+
+## `examples/` Directory Structure
+
+Examples in this the `examples/` directory follow a specific directory structure that allows Saturn to automatically seed them in a new user's environment with these.
+
+Each directory below `examples/` corresponds to one new Saturn Project + Jupyter that will be created in users' environments. The names of these directories should have names which only contain lowercase alphanumeric characters and dashes.
+
+```text
+.
+├── examples
+│   ├── taxi-cpu
+│   │   ├── dashboard
+│   │   │   └── dashboard.ipynb
+│   │   ├── etl
+│   │   │   └── data-loading.ipynb
+│   │   ├── README.md
+│   │   └── .saturn
+│   │       └── saturn.json
+│   │       └── start
+```
+
+### The `README.md`
+
+Each folder exactly one level below `examples/` should have a `README.md`. This should contain relevant information for understanding the example, such as:
+
+* description of any manual steps needed to use the code (like configuring credentials)
+* links to reference material like blogs posts or data dictionaries
+
+So, for example, in the directory structure above, `taxi-cpu/README.md` is required but `taxi-cpu/dashboard/README.md` is not.
+
+### The `.saturn` Directory
+
+Each example must contain a special directory called `.saturn`. This contains the details needed for Saturn to create the necessary resources for the example. This can contain the following files, and any other files will be ignored.
+
+* [`.saturn/saturn.json`](#saturn-json)
+* [`.saturn/start`](#start-script)
+
+#### `.saturn/saturn.json` <a name="saturn-json"></a>
+
+A JSON with the following structure:
+
+```json
+{
+    "image": "saturncloud/saturn:2020.07.08.1",
+    "jupyter": {
+        "size": "8xlarge",
+        "disk_space": "256Gi",
+        "ssh_enabled": false
+    },
+    "environment_variables": {
+        "TAXI_S3": "s3://saturn-titan/nyc-taxi"
+    }
+}
+```
+
+* `image`: the name of an image to use for the Jupyter server and all Dask clusters
+* `jupyter`: customization specific to the jupyter server, including:
+    - `size`: A valid size for a Saturn instance
+    - `disk_space`: An amount of disk space, in units `Gi`. See the "Disk Space" dropdown on the "Jupyter" page in the Saturn UI for valid values.
+    - `ssh_enabled`: A boolean indicating whether to set up [SSH access from outside Saturn into the Server](https://www.saturncloud.io/docs/connecting/tools/ssh/)
+* `environment_variables`:
+    - A dictionary whose keys are the names of environment variables, and whose values are the values for the environment variables.
+
+Any customization of Dask clusters should be done in notebook code, using [`dask-saturn`](https://github.com/saturncloud/dask-saturn).
+
+#### `.saturn/start` <a name="start-script"></a>
+
+A shell script that will be run on startup of the Jupyter server and all Dask resources. This should contain code that runs quickly, like `pip install`-ing libraries.
```

Comment:
Should this maybe just be included in the json? I don't have a strong preference either way.

Reply:
What would the advantage of that be?

I think that the start script should be a script that looks exactly like what you would put in the `Start Script` box in the UI. Otherwise there will be extra mental effort to figure out how the things in the repo relate to the things you see on the "Jupyter" page.

Reply:
The same could be said of environment_variables right?

Reply:
it could. The difference with environment variables is that they'll always be exactly one thing: a set of string keys and string values. The order they're defined is also irrelevant, so they don't depend on each other.

Those things aren't true for the `start` script. The `start` script can be any code, and I think that we'd inevitably end up stitching together some parts of it from `saturn.json` plus some parts that are a plain script if we added a `pip_dependencies` section in `saturn.json`.

Reply:
fair enough :)